python/phylanx/util/__init__.py

```diff
@@ -56,6 +77,39 @@ def recompile(self,a,allowreturn=False):
             else:
                 raise Exception()
             return s
+        elif nm == "Subscript":
+            e = get_node(a,name="ExtSlice")
+            s0 = get_node(e,name="Slice",num=0)
+            s1 = get_node(e,name="Slice",num=1)
+            xlo = s0.lower
+            xhi = s0.upper
+            ylo = s1.lower
+            yhi = s1.upper
+            sname = self.recompile(get_node(a,num=0))
+            s = "slice("
+            s += sname
+            s += ","
+            if xlo == None:
+                s += "0"
+            else:
+                s += self.recompile(xlo)
+            s += ","
+            if xhi == None:
+                s += "shape(" + sname + ",0)"
```

Comment:
You could use `-1` as the index to refer to the last row.

Reply:
The -1 in the last position should cut off the last argument, not show the whole array.

Reply:
Yes, you're certainly right.